x/gamm/pool-models/internal/cfmm_common/lp_test.go

```diff
@@ -0,0 +1,138 @@
+package cfmm_common_test
+
+import (
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/osmosis-labs/osmosis/v7/x/gamm/pool-models/balancer"
+	"github.com/osmosis-labs/osmosis/v7/x/gamm/pool-models/internal/cfmm_common"
+	"github.com/osmosis-labs/osmosis/v7/x/gamm/pool-models/stableswap"
+	gammtypes "github.com/osmosis-labs/osmosis/v7/x/gamm/types"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+// a helper function used to multiply coins
+func mulCoins(coins sdk.Coins, multiplier sdk.Dec) sdk.Coins {
+	outCoins := sdk.Coins{}
+	for _, coin := range coins {
+		outCoin := sdk.NewCoin(coin.Denom, multiplier.MulInt(coin.Amount).TruncateInt())
+		if !outCoin.Amount.IsZero() {
+			outCoins = append(outCoins, outCoin)
+		}
+	}
+	return outCoins
+}
+
+func TestCalcExitPool(t *testing.T) {
+	var (
+		emptyContext = sdk.Context{}
+
+		// create these pools used for testing
+		twoAssetPool, err1 = stableswap.NewStableswapPool(
+			1,
+			stableswap.PoolParams{ExitFee: sdk.ZeroDec()},
+			sdk.NewCoins(sdk.NewInt64Coin("foo", 1000000000),
+				sdk.NewInt64Coin("bar", 1000000000)),
+			"",
+			time.Now(),
+		)
+		threeAssetPool, err2 = balancer.NewBalancerPool(
+			1,
+			balancer.PoolParams{SwapFee: sdk.ZeroDec(), ExitFee: sdk.ZeroDec()},
+			[]balancer.PoolAsset{
+				{Token: sdk.NewInt64Coin("foo", 2000000000), Weight: sdk.NewIntFromUint64(5)},
+				{Token: sdk.NewInt64Coin("bar", 3000000000), Weight: sdk.NewIntFromUint64(5)},
+				{Token: sdk.NewInt64Coin("baz", 4000000000), Weight: sdk.NewIntFromUint64(5)},
```

Comment:
Seems that pool assets have the same values between the version with fee and the one without. Maybe we extract a `threePoolAssets` variable to avoid duplication?

Reply:
same with `twoPoolAssets`, assuming those are also duplicates